### PR TITLE
Add WithSSHConfig and PublicKeyLookupFunc

### PR DESCRIPTION
**Pull request by pjbgf**
This replaces the original gitkit fork with github.com/fluxcd/gitkit.
Having the capability of setting `ssh.ServerConfig` amongst other things,
enables the use of unit tests to reproduce scenarios that otherwise can
only be experienced in E2E tests.

### CODE REVIEW

**Comment by stefanprodan** [comment]
I’m for making an official fork on gitkit in fluxcd org.

**Comment by stefanprodan** [comment]
Fork here https://github.com/fluxcd/gitkit

File: gittestserver/go.mod [hunk 1]
```diff
@@ -6,8 +6,12 @@ require (
 	github.com/go-git/go-billy/v5 v5.3.1
 	github.com/go-git/go-git/v5 v5.4.2
 	github.com/sosedoff/gitkit v0.3.0
+	golang.org/x/crypto v0.0.0-20220427172511-eb4f295cb31f
 )
 
+// Replaces gitkit with fluxcd's fork, which extends the server settings that can be configured.
+replace github.com/sosedoff/gitkit => github.com/fluxcd/gitkit v0.4.0
```
Comment (stefanprodan) [inline gittestserver/go.mod:13]: I would use `fluxcd/gitkit` directly, if we manage to port all our changes to upstream then we can roll back to sosedoff.

Comment (pjbgf) [reply gittestserver/go.mod]: Sure, I have renamed the fork and updated the reference here. PTAL

**Review by stefanprodan — APPROVED** [review]
LGTM 

Thanks @pjbgf

**Pull request merged**